cfgrammar/src/lib/newlinecache.rs

```diff
@@ -135,17 +135,27 @@ impl FromStr for NewlineCache {
     }
 }
 
+impl<'a> FromIterator<&'a str> for NewlineCache {
+    fn from_iter<I>(iter: I) -> Self
+    where
+        I: IntoIterator<Item = &'a str>,
+    {
+        let mut nlcache = NewlineCache::new();
+        for s in iter.into_iter() {
+            nlcache.feed(s)
+        }
+        nlcache
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::NewlineCache;
+    use std::ops::Deref;
 
     fn newline_test_helper(feed: &[&str], tests: &[(usize, usize)]) {
-        let mut cache = NewlineCache::new();
-        let mut full_string = String::new();
-        for f in feed {
-            cache.feed(f);
-            full_string.push_str(f);
-        }
+        let cache: NewlineCache = feed.iter().map(|s| s.deref()).collect();
```

Comment:
I guess `feed.iter().copied().collect()` might be preferred?

Reply:
Yes, `copied` is Clippy-friendly, so let's do that.

Reply:
Alright, I pushed that, and squashed.

Reply:
OK. Though please don't squash without the reviewer asking next time, as it makes reviewing in general very difficult. This one time, fortunately, even my pea brain could cope with things :)

Reply:
Ok, I thought the intermediate commit would help, but will avoid that in the future.